airflow/contrib/operators/gcp_sql_operator.py

```diff
@@ -0,0 +1,299 @@
+        https://cloud.google.com/sql/docs/mysql/admin-api/v1beta4/instances/insert
+        #request-body
+        Caution: The body should always be as specified in insert, even if the
+        operation will be update. The minimal differences between them are handled by
+        this operator.
+    :type body: dict
+    :param name: Name of the Cloud SQL instance. This does not include the project ID.
+    :type name: str
+    :param gcp_conn_id: The connection ID used to connect to Google Cloud Platform.
+    :type gcp_conn_id: str
+    :param api_version: API version used (e.g. v1).
+    :type api_version: str
+    """
+    # [START gcp_sql_deploy_template_fields]
+    template_fields = ('project_id', 'name', 'gcp_conn_id', 'api_version',
+                       'validate_body')
+    # [END gcp_sql_deploy_template_fields]
+
+    @apply_defaults
+    def __init__(self,
+                 project_id,
+                 body,
+                 name,
+                 gcp_conn_id='google_cloud_default',
+                 api_version='v1beta4',
+                 validate_body=True,
+                 *args, **kwargs):
+        self.body = body
+        self.validate_body = validate_body
+        super(CloudSqlInstanceDeployOperator, self).__init__(
+            project_id=project_id, name=name, gcp_conn_id=gcp_conn_id,
+            api_version=api_version, *args, **kwargs)
+
+    def _validate_inputs(self):
+        super(CloudSqlInstanceDeployOperator, self)._validate_inputs()
+        if not self.body:
+            raise AirflowException("The required parameter 'body' is empty")
+
+    def _validate_body_fields(self):
+        if self.validate_body:
+            GcpBodyFieldValidator(CLOUD_SQL_VALIDATION,
+                                  api_version=self.api_version).validate(self.body)
+
+    def execute(self, context):
+        settings_version = self._check_if_instance_exists(self.name)
+        self._validate_body_fields()
+        if not settings_version:
+            # Instance doesn't exist: INSERT
+            return self._hook.insert_instance(self.project_id, self.body)
+        else:
+            # Instance exists: UPDATE
+            self._inject_settings_version_to_body(self.body, settings_version)
+            return self._hook.update_instance(self.project_id, self.body, self.name)
+
+    @staticmethod
+    def _inject_settings_version_to_body(body, settings_version):
+        body.get(SETTINGS)[SETTINGS_VERSION] = settings_version
+
+
+class CloudSqlInstancePatchOperator(CloudSqlBaseOperator):
+    """
+    Updates settings of a Cloud SQL instance.
+
+    Caution: This is a partial update, so only included values for the settings will be
+    updated, and the rest will remain unchanged. For a full update,
+    use CloudSqlInstanceDeployOperator.
+
+    In the request body, supply the relevant portions of an instance resource, according
+    to the rules of patch semantics.
```

Comment:
I don't think there's a single rule of patch semantics (I know this came from CloudSQL docs, but I find that sentence too vague).

Let's link to https://cloud.google.com/sql/docs/mysql/admin-api/how-tos/performance#patch right after this sentence.

Reply:
You're right, taken out of the context of the Cloud SQL docs this sentence sounded a bit weird here.
I've added the link.